kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
I may be wrong, but line 315 (`self.select_node(node)`) may need to be removed instead of the the changes in this pr, for getting the desired effect of being able to deselect the last selected item, when multiselect is False.